### DISCUSSION TITLE
Laravel pagination security

### DISCUSSION THREAD

**Discussion by yasseraugustus** [laravel/framework#41817] [Q&A]
# Do I need to validate the pagination parameters (page, perPage..)??

- while writing some code with Laravel api I searched if there is any validation via Laravel code but it does not exists!! so do I need to take care of it?

```
public function index(PaginationRequest $request){
       ....
}
```

**Comment by AlexWinder** [comment]
There is validation on the `page` value:

https://github.com/laravel/framework/blob/78eb4dabcc03e189620c16f436358d41d31ae11f/src/Illuminate/Pagination/Paginator.php#L58

With regards to the `perPage` this is set based on your code and so you would validate your own values to send into the paginator.

https://laravel.com/docs/9.x/pagination#paginating-eloquent-results
